lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -602,9 +602,8 @@ struct StatementVisitor
                                     mlir::raw_indented_ostream &stream,
                                     ValueMap &valueMap) {
     // Ignore operations which are handled in the Expression Visitor.
-    if (isa<smt::Int2BVOp>(op))
-      return op->emitError(
-          "int2bv operations are not supported for SMTLIB emission");
+    if (isa<smt::Int2BVOp, BV2IntOp>(op))
+      return op->emitError("operation not supported for SMTLIB emission");
 
     return success();
```

Comment:
I think just
```
  return op->emitError("operation not supported for SMTLIB emission");
```
should work instead of
```
    if (isa<smt::Int2BVOp, BV2IntOp>(op))
      return op->emitError("operation not supported for SMTLIB emission");

    return success();
```
to generically emit this error if we add new operations that we don't explicitly add support for in ExportSMTLIB (i.e., should be more robust)

Reply:
Ah I see, sorry for the misunderstanding before!

Reply:
Just checked and this actually breaks all the ExportSMTLIB tests - I'm not super clear on the Visitor infrastructure so I'm not sure why yet, but from a brief scan it looks like getting rid of the condition causes the error to throw on the first non-ConstantLike op the export pass encounters.